### PR TITLE
add retries to flaky test HeliosSoloIT.testServiceDiscovery

### PR DESCRIPTION
**Pull request by mattnworb**
The test can fail because it tries to connect immediately to the socket of
the alpine container. Need to give the `apk install` being run as the
command of the container time to finish.

### CODE REVIEW

**Comment by mattnworb** [comment]
@spotify/helios-team

File: helios-integration-tests/src/test/java/com/spotify/helios/HeliosSoloIT.java [hunk 1]
```diff
@@ -114,10 +118,15 @@ public void testServiceDiscovery() throws Exception {
 
     // Connect to alpine container to get the curl response. If we get back the nginx welcome page
     // we know that helios properly registered the nginx service in SkyDNS.
-    try (final Socket s = new Socket(alpineAddress.getHostText(), alpineAddress.getPort())) {
-      final String result = IOUtils.toString(s.getInputStream()).trim();
-      assertThat(result, containsString("Welcome to nginx!"));
-    }
+    final Callable<String> curlResponse = () -> {
```
Comment (davidxia) [inline helios-integration-tests/src/test/java/com/spotify/helios/HeliosSoloIT.java]: Perhaps just call this var `response` since it's not curl?

**Comment by davidxia** [comment]
:100: if it works

File: helios-integration-tests/src/test/java/com/spotify/helios/HeliosSoloIT.java (hunk 1 above)
Comment (mattnworb) [inline helios-integration-tests/src/test/java/com/spotify/helios/HeliosSoloIT.java]: i was trying to use a verb, but I'll find a better one :)

**Comment by mattnworb** [comment]
The test failure is a different flaky test so I'll merge this fix.

**Pull request merged**